addons/project_purchase/models/purchase_order_line.py

```diff
@@ -1,15 +1,13 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import models
 
 
 class PurchaseOrderLine(models.Model):
     _inherit = 'purchase.order.line'
 
-    def _default_account_analytic_id(self):
+    def _compute_analytic_distribution(self):
+        super()._compute_analytic_distribution()
         if self._context.get('project_id'):
-            return self.env['project.project'].browse(self._context['project_id']).analytic_account_id
-        return False
-
-    account_analytic_id = fields.Many2one(default=_default_account_analytic_id)
+            self.analytic_distribution = {self.env['project.project'].browse(self._context['project_id']).analytic_account_id: 100}
```

Comment:
Shouldn't we have the same mechanism as in sale to prevent it being recomputed if it is linked to a project?

Reply:
Even if we want, we can't, as we don't have the fields specifying that an aml comes from a purchase_order_line (if I'm not mistaken atleast)